app/org/maproulette/cache/CaffeineCache.scala

```diff
@@ -0,0 +1,139 @@
+package org.maproulette.cache
+
+import com.github.blemale.scaffeine.Scaffeine
+import org.maproulette.Config
+import play.api.Logging
+
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.FiniteDuration
+
+class CaffeineCache[Key, Value <: CacheObject[Key]](config: Config)
+    extends Cache[Key, Value]
+    with Logging {
+  override implicit val cacheLimit: Int  = config.cacheLimit
+  override implicit val cacheExpiry: Int = config.cacheExpiry
+  val caffeineCache: com.github.blemale.scaffeine.Cache[Key, BasicInnerValue[Key, Value]] =
+    Scaffeine()
+      .recordStats()
+      .expireAfterAccess(FiniteDuration(config.cacheExpiry, TimeUnit.SECONDS))
+      .maximumSize(config.cacheLimit)
+      .build[Key, BasicInnerValue[Key, Value]]()
+
+  /**
+    * Checks if an item is cached or not
+    *
+    * @param key The id of the object to check to see if it is in the cache
+    * @return true if the item is found in the cache
+    */
+  override def isCached(key: Key): Boolean = caffeineCache.getIfPresent(key).nonEmpty
+
+  /**
+    * Fully clears the cache, this may not be applicable for non basic in memory caches
+    */
+  override def clear(): Unit = caffeineCache.invalidateAll()
+
+  /**
+    * @return the current size of the cache
+    */
+  override def size(): Long = caffeineCache.estimatedSize()
+
+  /**
+    * Adds an object to the cache, if cache limit has been reached, then will remove the oldest
+    * accessed item in the cache
+    *
+    * @param key   The object to add to the cache
+    * @param value You can add a custom expiry to a specific element in seconds
```

Comment:
fyi, you can use a custom expiration policy, `expireAfter(Expiry)`, if you want per-entry evaluations.

Reply:
Thanks for the comment and your work on caffeine! It's pretty great :+1:  

Within [maproulette](https://maproulette.org/) the custom expiration doesn't appear to be used (except in a test) and the param should probably be eliminated, so I just ignored it for now